src/layouts/MainLayout.astro

```diff
@@ -55,5 +55,27 @@ const { ...head } = Astro.props
 				overflow-y: auto;
 			}
 		</style>
+		<script>
+			import listen from "micromorph/nav"
+			const cache = new Set()
+			listen({
+				beforeDiff(doc) {
+					const promises = []
+					for (const { href } of doc.querySelectorAll('link[rel="stylesheet"]')) {
+						if (cache.has(href)) continue
+
+						promises.push(
+							fetch(new URL(href, window.location.toString())).then((res) => {
+								cache.add(href)
+							}),
+						)
+					}
+					return Promise.race([
+						new Promise((resolve) => setTimeout(resolve, 100)),
```

Comment:
what is all of this??? the animation is so nice tho

Reply:
This is a hacky way to preload the incoming CSS, if there is any

Reply:
Removed! Not needed.